src/source-napatech.c

```diff
@@ -76,52 +72,63 @@ TmEcode NoNapatechSupportExit(ThreadVars *tv, const void *initdata, void **data)
 
 #else /* Implied we do have NAPATECH support */
 
+
 #include <nt.h>
 
+#define MAX_STREAMS 256
+
 extern int max_pending_packets;
 
 typedef struct NapatechThreadVars_ {
     ThreadVars *tv;
     NtNetStreamRx_t rx_stream;
-    uint64_t stream_id;
+    uint16_t stream_id;
     int hba;
-    uint64_t pkts;
-    uint64_t drops;
-    uint64_t bytes;
-
     TmSlot *slot;
 } NapatechThreadVars;
 
 
 TmEcode NapatechStreamThreadInit(ThreadVars *, const void *, void **);
 void NapatechStreamThreadExitStats(ThreadVars *, void *);
-TmEcode NapatechStreamLoop(ThreadVars *tv, void *data, void *slot);
+TmEcode NapatechPacketLoopZC(ThreadVars *tv, void *data, void *slot);
 
 TmEcode NapatechDecodeThreadInit(ThreadVars *, const void *, void **);
 TmEcode NapatechDecodeThreadDeinit(ThreadVars *tv, void *data);
 TmEcode NapatechDecode(ThreadVars *, Packet *, void *, PacketQueue *, PacketQueue *);
 
+// These are used as the threads are exiting to get a comprehensive count of
+// all the packets received and dropped.
+SC_ATOMIC_DECLARE(uint64_t, total_packets); 
```

Comment:
I think you can also use the regular stats api here. Should be more efficient that using atomics.